packages/vite/src/node/optimizer/scan.ts

```diff
@@ -534,11 +534,8 @@ function extractImportPaths(code: string) {
 
   let js = ''
   let m
+  importsRE.lastIndex = 0
   while ((m = importsRE.exec(code)) != null) {
-    // This is necessary to avoid infinite loops with zero-width matches
-    if (m.index === importsRE.lastIndex) {
-      importsRE.lastIndex++
-    }
```

Comment:
Could you explain why this isnt needed any more?

Reply:
This is used to safe-guard zero-width matches, e.g. the regex would match `""`. But since `importsRE` is guaranteed to never do that (it always captures some text), this can be removed.

I think this part was copied from https://regex101.com generated output before.